src/engine/service/runner/backend/docker.rs

```diff
@@ -62,35 +61,43 @@ impl Runner {
                 container_create(&name, execution, &mut client).await;
                 container_start(&name, &mut client).await;
 
-                let mut logs = configure_logs(&name, execution, &mut client);
+                let logs = configure_logs(&name, execution, &mut client);
                 let mut wait = configure_wait(&name, &mut client);
 
-                let status;
-                let mut stdout = String::with_capacity(1 >> 8);
-                let mut stderr = String::with_capacity(1 >> 8);
-
-                loop {
-                    select! {
-                        result = logs.next().fuse() => match result {
-                            Some(Ok(LogOutput::StdOut { message })) => {
-                                stdout.push_str(&String::from_utf8_lossy(message.as_ref()))
-                            }
-                            Some(Ok(LogOutput::StdErr { message })) => {
-                                stderr.push_str(&String::from_utf8_lossy(message.as_ref()))
+                // Process logs until they stop when container stops
+                let (stdout, stderr) = logs
+                    .try_fold(
+                        (String::new(), String::new()),
```

Comment:
I would recommend initializing these with capacities—you're almost positive you're going to get something here, so skipping the first few grow operations would be good. I had selected `1 >> 8`, but you could do anything that makes sense to you.